driver-core/src/test/java/com/datastax/driver/core/ZeroTokenNodesIT.java

```diff
@@ -164,19 +164,26 @@ public void should_discover_zero_token_DC_when_option_is_enabled(
       // Later on we may want to adjust the LBP behavior to be more sophisticated.
       session.execute("DROP KEYSPACE IF EXISTS ZeroTokenNodesIT");
       session.execute(
-          "CREATE KEYSPACE ZeroTokenNodesIT WITH replication = {'class': 'NetworkTopologyStrategy', 'replication_factor': 2};");
+          "CREATE KEYSPACE ZeroTokenNodesIT WITH replication = {'class': 'NetworkTopologyStrategy', 'dc1': 2, 'dc2': 0};");
       session.execute("CREATE TABLE ZeroTokenNodesIT.tab (id INT PRIMARY KEY)");
+      PreparedStatement ps = session.prepare("INSERT INTO ZeroTokenNodesIT.tab (id) VALUES (?)");
       Set<InetSocketAddress> queriedNodes = new HashSet<>();
       for (int i = 0; i < 30; i++) {
-        ResultSet rs = session.execute("INSERT INTO ZeroTokenNodesIT.tab (id) VALUES (" + i + ")");
+        ResultSet rs = session.execute(ps.bind(i).setConsistencyLevel(ConsistencyLevel.TWO));
         queriedNodes.add(rs.getExecutionInfo().getQueriedHost().getEndPoint().resolve());
       }
-      assertThat(queriedNodes)
-          .containsExactly(
-              ccmBridge.addressOfNode(1),
-              ccmBridge.addressOfNode(2),
-              ccmBridge.addressOfNode(3),
-              ccmBridge.addressOfNode(4));
+
+      if (isDcAware) {
+        assertThat(queriedNodes)
+            .containsExactly(ccmBridge.addressOfNode(1), ccmBridge.addressOfNode(2));
+      } else {
+        assertThat(queriedNodes)
+            .containsExactly(
+                ccmBridge.addressOfNode(1),
+                ccmBridge.addressOfNode(2),
+                ccmBridge.addressOfNode(3),
+                ccmBridge.addressOfNode(4));
```

Comment:
It should ignore zero token nodes, no zero token nodes should be part of execution plan

Reply:
Why though?

Reply:
I thought that the option to allow them in was for that

Reply:
Oh, sorry i missed that place.